contracts/0.8.9/StakingRouter.sol

```diff
@@ -0,0 +1,570 @@
+
+    uint256 public constant DEPOSIT_SIZE = 32 ether;
+
+    uint256 public constant TOTAL_BASIS_POINTS = 10000;
+
+    /// @dev total count of staking modules
+    uint256 private _stakingModulesCount;
+
+    /// @dev id of the last added staking module. This counter grow on staking modules adding
+    uint24 private _lastStakingModuleId;
+
+    /// @dev mapping is used instead of array to allow to extend the StakingModule
+    mapping(uint256 => StakingModule) private _stakingModules;
+
+    /// @dev Position of the staking modules in the `_stakingModules` map, plus 1 because
+    ///      index 0 means a value is not in the set.
+    mapping(uint24 => uint256) private _stakingModuleIndicesOneBased;
+
+    constructor(address _depositContract) BeaconChainDepositor(_depositContract) {}
+
+    function initialize(address _admin, bytes32 _withdrawalCredentials) external {
+        if (_admin == address(0)) revert ErrorZeroAddress("_admin");
+        if (CONTRACT_VERSION_POSITION.getStorageUint256() != 0) revert ErrorBaseVersion();
+
+        _setupRole(DEFAULT_ADMIN_ROLE, _admin);
+
+        CONTRACT_VERSION_POSITION.setStorageUint256(1);
+        WITHDRAWAL_CREDENTIALS_POSITION.setStorageBytes32(_withdrawalCredentials);
+
+        emit ContractVersionSet(1);
+        emit WithdrawalCredentialsSet(_withdrawalCredentials);
+    }
+
+    /// @dev prohibit direct transfer to contract
+    receive() external payable {
+        revert ErrorDirectETHTransfer();
+    }
+
+    /**
+     * @notice register a new module
+     * @param _name name of module
+     * @param _stakingModuleAddress target percent of total keys in protocol, in BP
+     * @param _targetShare target total stake share
+     * @param _moduleFee fee of the module taken from the consensus layer rewards
+     * @param _treasuryFee treasury fee
+     */
+    function addModule(
+        string memory _name,
+        address _stakingModuleAddress,
+        uint16 _targetShare,
+        uint16 _moduleFee,
+        uint16 _treasuryFee
+    ) external onlyRole(MODULE_MANAGE_ROLE) {
+        if (_targetShare > TOTAL_BASIS_POINTS) revert ErrorValueOver100Percent("_targetShare");
+        if (_treasuryFee > TOTAL_BASIS_POINTS) revert ErrorValueOver100Percent("_treasuryFee");
+        if (_moduleFee > TOTAL_BASIS_POINTS) revert ErrorValueOver100Percent("_moduleFee");
+
+        uint24 newStakingModuleId = _lastStakingModuleId + 1;
+        uint256 newStakingModuleIndex = _stakingModulesCount;
+        StakingModule storage newStakingModule = _stakingModules[newStakingModuleIndex];
+
+        newStakingModule.id = newStakingModuleId;
+        newStakingModule.name = _name;
+        newStakingModule.stakingModuleAddress = _stakingModuleAddress;
+        newStakingModule.targetShare = _targetShare;
+        newStakingModule.treasuryFee = _treasuryFee;
+        newStakingModule.moduleFee = _moduleFee;
+        newStakingModule.status = uint8(StakingModuleStatus.Active);
```

Comment:
Let's have a comment about uint8 and enum

Reply:
added